app/controllers/samples_controller.rb

```diff
@@ -157,7 +157,7 @@ def show
     @project_info = @sample.project ? @sample.project : nil
     @project_sample_ids_names = @sample.project ? Hash[current_power.project_samples(@sample.project).map { |s| [s.id, s.name] }] : nil
     @host_genome = @sample.host_genome ? @sample.host_genome : nil
-    @background_models = Background.all
+    @background_models = current_power.backgrounds
```

Comment:
since you are adding access control related code. unit test is required. Please add a few test cases to 
- make sure users who are allowed access have access
- make sure users who are not allowed access won't access it.

Reply:
You will also need to make sure people don't access individual background through report_info, so make sure you use current_power.backgrounds.find in the code.